app/parse.py

```diff
@@ -8,5 +11,30 @@ class Course:
     duration: str
 
 
-def get_all_courses() -> list[Course]:
-    pass
+BASE_URL = "https://mate.academy/"
+
+
+def parse_single_course(course: Tag) -> Course:
+    return Course(
+        name=course.find("h3").text,
+        short_description=course.select(
+            ".typography_landingTextMain__Rc8BD"
+        )[-1].text,
+        duration=course.select("span")[-2].text,
+    )
+
+
+def get_soup_of_page(url: str, session: requests.Session) -> BeautifulSoup:
+    response = session.get(url)
+    return BeautifulSoup(response.content, features="html.parser")
+
+
+def get_all_courses() -> list[object]:
```

Comment:
The return type of the `get_all_courses` function should be `list[Course]` instead of `list[object]` to reflect that it returns a list of `Course` objects. This will improve type safety and clarity.